src/daemon/rpmostreed-sysroot.c

```diff
@@ -732,10 +750,92 @@ sysroot_constructed (GObject *object)
   G_OBJECT_CLASS (rpmostreed_sysroot_parent_class)->constructed (object);
 }
 
+static gboolean
+sysroot_authorize_method (GDBusInterfaceSkeleton *interface,
+                          GDBusMethodInvocation  *invocation)
+{
+  RpmostreedSysroot *self = RPMOSTREED_SYSROOT (interface);
+  const gchar *method_name = g_dbus_method_invocation_get_method_name (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  gboolean authorized = FALSE;
+  const char *action = NULL;
+
+  if (self->on_session_bus)
+    {
+      /* The daemon is on the session bus, running self tests */
+      authorized = TRUE;
+    }
+  else if (g_strcmp0 (method_name, "GetOS") == 0)
+    {
+      /* GetOS() is always allowed */
+      authorized = TRUE;
+    }
+  else if (g_strcmp0 (method_name, "RegisterClient") == 0 ||
+           g_strcmp0 (method_name, "UnregisterClient") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.client-management";
```

Comment:
I guess I was thinking of a potential use case for a background service running in a service account that runs e.g. `Deploy`. Then, you could add rules to allow that uid. I'm fine with dropping that for now until we flesh that out that story more though.